ci/docker/linux-apt-r.dockerfile

```diff
@@ -100,7 +100,7 @@ ENV \
     ARROW_BUILD_STATIC=OFF \
     ARROW_BUILD_TESTS=OFF \
     ARROW_BUILD_UTILITIES=OFF \
-    ARROW_DEPENDENCY_SOURCE=SYSTEM \
+    ARROW_DEPENDENCY_SOURCE="" \
```

Comment:
Why this change? This build should not be affected by the scope of the changes in this PR.

Reply:
This is that stacking of docker images + env vars I mentioned earlier. For any of our test jobs that use the `ubuntu-r` (and a few other linux distros) docker compose service, they will set the env var `ARROW_DEPENDENCY_SOURCE` to `SYSTEM` (e.g. https://github.com/apache/arrow/blob/master/ci/docker/ubuntu-21.04-cpp.dockerfile#L109). This is true even of our `ubuntu-r-only-r` task that extends (docker's term, not mine) the `ubuntu-r` service to *not* build libarrow first and only rely on the R package process to build libarrow. So all of those tests were no longer testing environments that most people would experience (with the `ARROW_DEPENDENCY_SOURCE` env var unset). 

I'll try moving this to the `environment` of the `ubuntu-r-only-r` service, since that's slightly more targeted and should work. But for a CI job that is intended to mimic what people are doing when they are installing (which was the original intention behind the `test-r-ubuntu-21.04` task, I think we want this to install with that env var unset.